test/pages/Home.test.jsx

```diff
@@ -1,24 +1,30 @@
-import { render, cleanup } from '@testing-library/react';
 import React from 'react';
+import { render, cleanup, screen } from '@testing-library/react';
 import { expect, it, afterEach, describe } from 'vitest';
-import { ThemeProvider } from '@mui/material/styles';
+import { Home } from '@pages';
+import createMatchMedia from '../helpers/createMatchMedia';
 
-import Home from '../../src/pages/Home';
-import theme from '../../src/theme';
+describe('Home Page', () => {
+  afterEach(cleanup);
+  // if tests are failing check this value vs value in Home.jsx
+  const h1FontSize = '144px';
```

Comment:
Same as above

Reply:
This is the same thing.  This is the current base value - so .toBe(basevalue) & other tests expect .not.toBe(basevalue).

Reply:
I'd rather it be something like this, so I can check the ordered of the rendered elements:

```js
it('renders mobile', () => {
    window.matchMedia = createMatchMedia(599);
    render(<Home />);
    const h1 = screen.getByTestId('testStack');
    const { children } = h1;
    
    expect(children[0].toEqual(<Typography />));
  });
  ```
  
  then desktop would just be like `expect(children[0].not.toEqual(<Typography />));`
  
  but can't seem to figure out syntax or where in docs to find much besides the matchers.

Reply:
Ok this seemingly works for the intention:

```js
describe('Home Page', () => {
  afterEach(cleanup);

  it('renders with correct order of logoSection on mobile', () => {
    window.matchMedia = createMatchMedia(599);
    render(<Home />);
    const h1 = screen.getByTestId('testStack');
    const children = Array.from(h1.children);

    expect(children.length).toBe(3);
    expect(children[0].tagName.toLowerCase()).toBe('span');
    expect(children[1].tagName.toLowerCase()).toBe('img');
    expect(children[2].tagName.toLowerCase()).toBe('span');
  });

  it('rrenders with correct order of logoSection on desktop', () => {
    window.matchMedia = createMatchMedia(1200);
    render(<Home />);
    const h1 = screen.getByTestId('testStack');
    const children = Array.from(h1.children);

    expect(children.length).toBe(2);
    expect(children[0].tagName.toLowerCase()).toBe('span');
    expect(children[1].tagName.toLowerCase()).toBe('span');
  });
});
```

Reply:
This to me still feels very fragile. We're testing the exact order of the DOM, when for the user, none of this matters. If the goal is to ensure that the DOM doesn't change, let's just rely on snapshots as that seems more accurate for the purpose. Otherwise, I think tests should be testing the behavior that the user will experience

Reply:
> This to me still feels very fragile. We're testing the exact order of the DOM, when for the user, none of this matters. If the goal is to ensure that the DOM doesn't change, let's just rely on snapshots as that seems more accurate for the purpose. Otherwise, I think tests should be testing the behavior that the user will experience

But the goal for this was not to ensure the DOM doesn't change. Quite the contrary.  In this particular component that is being tested the DOM should be different for mobile and desktop. This tests just that without the reliance on a css style check.

This is you h1 section that you figured out the nested syntax of.